INSTALL-docker.md

```diff
@@ -0,0 +1,35 @@
+# Installing the CircleCI Runner on a host with Docker
```

Comment:
Perhaps add a bit of preamble here about the execution environment, and how jobs re-use the same container